w06/test/gad/DualPivotQuicksortTest.java

```diff
@@ -0,0 +1,165 @@
+        int[] numbersCopy = Arrays.copyOf(numbers, numbers.length);
+        DualPivotQuicksort dualPivotQuicksort = new DualPivotQuicksort(DualPivotFinder.getFirstLastPivot(), 0);
+        StudentResult result = new StudentResult();
+
+        dualPivotQuicksort.sort(numbersCopy, result, 0, numbersCopy.length - 1);
+        assertArrayEquals(expected, numbersCopy);
+    }
+
+
+    //for this test to work you have to remove the "result.log()" from your DualPivotQuicksort sort() method
+    @Test
+    public void dualPivotQuickSortTestLarge() {
+        Random random = new Random();
+        int[] numbers = new int[1_000_000];
+        for (int i = 0; i < numbers.length; i++) {
+            numbers[i] = random.nextInt(Integer.MAX_VALUE);
+        }
+        int[] expected = Arrays.copyOf(numbers, numbers.length);
+
+        DualPivotQuicksort dualPivotQuicksort = new DualPivotQuicksort(DualPivotFinder.getRandomPivot(10), 0);
+        StudentResult result = new StudentResult();
+
+        long start = System.nanoTime();
+        Arrays.sort(expected);
+        long binaryTime = System.nanoTime() - start;
+
+        start = System.nanoTime();
+        dualPivotQuicksort.sort(numbers, result, 0, numbers.length - 1);
+        long decimalTime = System.nanoTime() - start;
+
+        assertArrayEquals(expected, numbers);
+        System.out.println("Your DualPivotQuicksort: " + binaryTime / 1_000_000);
+        System.out.println("Java sort: " + decimalTime / 1_000_000);
+    }
+
+    @RepeatedTest(1000)
+    public void dualPivotQuickSortTestPartial() {
+        int[] numbers = new int[]{5, 12, 8, 3, 9, 7, 2, 1, 6, 4, 10, 15, 11, 13, 14, 17, 19, 18, 20, 16, 23, 22, 21, 25, 24, 26, 29, 27, 30, 28, 33, 31, 34, 36, 32, 35, 38, 40, 37, 39, 44, 42, 41, 47, 45, 50, 46, 43, 49, 48};
+        int[] expected = Arrays.copyOf(numbers, numbers.length);
+        int[] expectedCopy = Arrays.copyOf(expected, expected.length);
+        int[] expectedCopy3 = new int[expectedCopy.length];
+
+        Random random = new Random();
+
+        int from = random.nextInt(numbers.length);
+        int to = random.nextInt(numbers.length);
+
+        if (from > to) {
+            int temp = from;
+            from = to;
+            to = temp;
+        }
+
+        int[] expectedSortedPart = new int[to - from + 1];
+
+        int j = 0;
+        for (int i = from; i <= to; i++) {
+            expectedSortedPart[j++] = expectedCopy[i];
+        }
+        Arrays.sort(expectedSortedPart);
+
+        for (int i = 0; i < expected.length; i++) {
+            if (i >= from && i <= to) {
+                expectedCopy3[i] = expectedSortedPart[i - from];
+            } else {
+                expectedCopy3[i] = expected[i];
+            }
+        }
+
+        DualPivotQuicksort dualPivotQuicksort = new DualPivotQuicksort(DualPivotFinder.getRandomPivot(10), 0);
+        StudentResult result = new StudentResultForTest();
```

Comment:
You have to change `StudentResult` to `StudentResultForTest` otherways you get a error.